modules/vf-graphql-holochain/mutations/proposedIntent.ts

```diff
@@ -5,26 +5,33 @@
  * @since:   2019-09-12
  */
 
-import { DNAIdMappings } from '../types'
+import { ByRevision, DNAIdMappings, IntentAddress, ProposalAddress } from '../types'
 import { mapZomeFn } from '../connection'
 import { deleteHandler } from './'
 
 import {
   ProposedIntentResponse,
 } from '@valueflows/vf-graphql'
 
+export interface CreateArgs {
+  proposedIntent: {
+    publishedIn: ProposalAddress,
+    publishes: IntentAddress,
+    reciprocal: boolean,
+  },
+}
 export type createHandler = (root: any, args) => Promise<ProposedIntentResponse>
 
 export default (dnaConfig: DNAIdMappings, conductorUri: string) => {
-  const runCreate = mapZomeFn(dnaConfig, conductorUri, 'proposal', 'proposed_intent', 'create_proposed_intent')
-  const runDelete = mapZomeFn(dnaConfig, conductorUri, 'proposal', 'proposed_intent', 'delete_proposed_intent')
+  const runCreate = mapZomeFn<CreateArgs, ProposedIntentResponse>(dnaConfig, conductorUri, 'proposal', 'proposed_intent', 'create_proposed_intent')
+  const runDelete = mapZomeFn<ByRevision, boolean>(dnaConfig, conductorUri, 'proposal', 'proposed_intent', 'delete_proposed_intent')
 
   const proposeIntent: createHandler = async (root, args) => {
-    return runCreate({ proposed_intent: args })
+    return runCreate({ proposedIntent: args })
```

Comment:
```suggestion
    return runCreate(args)
```
I think without reverting, this will be a bug. Add the typescript definition that's missing for `args` to the above `createHandler` type definition.

Reply:
I think my new changes are right, this is a bit weird because I think there is a bit of a discrepancy in input types for `create` mutations, if the mutation has a defined input type (any `CreateParams` object), vs multiple inputs and how gql handles this (I think?)

Reply:
ah ok. Let me review what you're saying about that.

Reply:
Here are two resources we can build off: 
https://www.apollographql.com/tutorials/fullstack-quickstart/writing-query-resolvers
https://graphql.org/graphql-js/passing-arguments/

Reply:
You're correct, we could follow up on this eventually potentially to make it the same as the others.